src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -186,7 +186,10 @@ public static void RemoveDirectory(string fullPath, bool recursive)
             }
 
             Interop.Kernel32.WIN32_FIND_DATA findData = default;
-            GetFindData(fullPath, isDirectory: true, ref findData);
+            // FindFirstFile($path) (used by GetFindData) fails with ACCESS_DENIED when user has no ListDirectory rights
+            // but FindFirstFile($path/*") (used by RemoveDirectoryRecursive) works fine in such scenario.
+            // So we ignore it here and let RemoveDirectoryRecursive throw if FindFirstFile($path/*") fails with ACCESS_DENIED.
```

Comment:
Do you know why `Path.Join(fullPath, "*")` is needed in `RemoveDirectoryRecursive`?

Reply:
My understanding is that `FindFirstFile($pathToDirectory)` returns info for given directory, while `FindFirstFile($pathToDirectory/*)` returns info for the first file in given directory (which is needed here as we want to recursively remove all files and subfolders)